#34 Set up Node.js REST API empty project

bryan936M opened the pull request:
#### What does this PR do?
- create the basic components of a restful api
#### Description of Task to be completed?
- add project structures components(model, routes, test, config, .github, etc) 
- add documentation(swagger)
- configure eslintr
#### How should this be manually tested?
- clone repo
- open terminal & run ' npm install' command
- again; execute 'npm run dev' command
- open browser & search 'http://localhost:3000/' & 'http://localhost:3000/api-docs'
#### Any background context you want to provide?
- set indent space of vscode editor to 2 spaces in settings under 'FILE: EOL' for eslint
- default port is 3000
#### What are the relevant pivotal tracker/Trello stories?
card Id: #34
card link: https://trello.com/c/PHkEazmg
#### Screenshots (if appropriate)

Commit:
trial

Commit:
chore(setup): create basic components

- create basic RESTFUL API structure
- add swagger documentation
- add & configure eslintr

[Delivers #34]

chanceDira commented on .eslintrc.js:
```diff
@@ -0,0 +1,17 @@
+    es2021: true,
+    node: true,
+  },
+  extends: 'airbnb-base',
```
Here let's also add "prettier"

chanceDira commented on app.js:
```diff
@@ -0,0 +1,60 @@
+// Imports
+const express = require('express');
```
can you use import instead of require

chanceDira commented on server.js:
```diff
@@ -0,0 +1,9 @@
+
+// Listen to port ( default: 3000 )
+app.listen(PORT, () => {
+  console.log('[Server] On');
```
Include the listening PORT

chanceDira reviewed (changes requested):
Good job @Bryan, just added some feedback for you

faid-terence reviewed (commented):
This is very good Bryan , i would also advice you to add a controller folder for better practices
otherwise all is Good, Thanks

Commit:
chore(setup): add feedback to branch

- modified require to import statement
- modified log statement in server to include port
- add prettier to eslint

[Delivers #34]

Commit:
chore(setup): create controller folder

- add controller folder to project
- add delete sample file

[Delivers #34]

mukunzidd reviewed (changes requested):
- Please remove the brackets in the title of the PR
- Link your troll card in the description. Not just a card id
- Add a lint script in package.json

Other than that, it looks good.

bryan936M committed:
chore(setup): apply feedback

- add eslint script to package.json

[Delivers feedback]

chanceDira reviewed (approved):
Good job @bryan

ALVINdimpos commented on package.json:
```diff
@@ -0,0 +1,44 @@
+    "swagger-jsdoc": "^6.2.8",
+    "swagger-ui-express": "^4.6.2"
+  },
+  "devDependencies": {
```
add this dependencies     "@babel/cli": "^7.21.0",
    "@babel/core": "^7.21.3",
    "@babel/preset-env": "^7.20.2",
    "jest": "^29.5.0"

bryan936M replied:
can you give me the names of the packages?

nili-alu replied:
npm install --save-dev @babel/cli
npm install --save-dev @babel/core
npm install @babel/preset-env --save-dev
npm install jest --global for jest,

ALVINdimpos replied:
thank you @nili-alu

Commit:
chore(setup): apply feedback

- add jest to package.json
- add babel/cli to package.json
- add babel/core to package.json
- add babel/preset to package.json

[Delivers feedback]

Commit:
Merge branch 'ch-project-setup-34' of https://github.com/atlp-rwanda/ecommerce-app-predators-bn into ch-project-setup-34

ALVINdimpos commented on package.json:
```diff
@@ -0,0 +1,48 @@
+    "url": "https://github.com/atlp-rwanda/ecommerce-app-predators-bn/issues"
+  },
+  "homepage": "https://github.com/atlp-rwanda/ecommerce-app-predators-bn#readme",
+  "dependencies": {
```
now it's good
